bindings/perl5/CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+if(NOT WITH_PERL5)
+    return()
+endif()
+
+
+message("Building bindings for perl5")
+
+
+find_package(Perl REQUIRED)
+find_package(PerlLibs REQUIRED)
+
+
+# Preprocessor flags (cppflags) are already included in compiler flags
+# (ccflags). Except a path to perl.h that is recorded nowhere.
+include_directories(${PERL_INCLUDE_PATH})
+perl_get_info(PERL_CFLAGS "ccflags")
+separate_arguments(PERL_CFLAGS)
+
+# remove options unused by clang
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    list(REMOVE_ITEM PERL_CFLAGS -fstack-clash-protection)
+    list(REMOVE_ITEM PERL_CFLAGS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1)
+    list(REMOVE_ITEM PERL_CFLAGS -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1)
```

Comment:
What happens when these aren't in the compiler flags? Also, shouldn't these be popped off globally as they'll fail to function in that circumstance everywhere?

Reply:
These don't work with clang:
clang-10: error: unknown argument: '-fstack-clash-protection'
clang-10: error: argument unused during compilation: '-specs=/usr/lib/rpm/redhat/redhat-hardened-cc1' [-Werror,-Wunused-command-line-argument]
clang-10: error: argument unused during compilation: '-specs=/usr/lib/rpm/redhat/redhat-annobin-cc1' [-Werror,-Wunused-command-line-argument]

I suppose handling these globally doesn't make sense, because these are only added by using PERL_CFLAGS when compiling the bindings. I suppose it's correct because Petr Pisar who wrote this understands Perl specifics more than we do.

Reply:
Okay, I'm just concerned about what happens when those parameters _don't_ exist from `PERL_CFLAGS`...

Reply:
I tried `list(REMOVE_ITEM PERL_CFLAGS -foo-bar)` outside the `if` block and cmake doesn't seem to complain.